### PR TITLE
Use multi-arch image for metrics server

### LINKED ISSUE #372: 'Exec format error' installing metrics server on k3s version v0.4.0 (7822549f) arm64 platform
**Describe the bug**
'Exec format error' installing metrics server on k3s version v0.4.0 (7822549f) arm64 platform

**To Reproduce**
```
curl -sfL https://get.k3s.io | INSTALL_K3S_EXEC="--disable-agent --kubelet-arg='address=0.0.0.0'" sh -

kubectl apply -f ./recipes/metrics-server/
```

**Expected behavior**
```
kubectl top node
```

**Additional context**
#252 #355

### PR DESCRIPTION
**Pull request by meldafrawi**
Fix #372

### CODE REVIEW

File: recipes/metrics-server/metrics-server-deployment.yaml [hunk 1]
```diff
@@ -36,7 +36,7 @@ spec:
         # - --metric-resolution=10s
         - --kubelet-insecure-tls
         - --kubelet-preferred-address-types=InternalIP
-        image: k8s.gcr.io/metrics-server-amd64:v0.3.1
+        image: k8s.gcr.io/metrics-server-arm:v0.3.1
```
Comment (erikwilson) [inline recipes/metrics-server/metrics-server-deployment.yaml]: We probably shouldn't change the default from amd64 to arm, the right thing to do is get metrics-server to use a multi-arch image.

Comment (meldafrawi) [reply recipes/metrics-server/metrics-server-deployment.yaml]: I've checked, and I couldn't find metric server multi-arch image, However i found a related opened pull request  [PR #182](https://github.com/kubernetes-incubator/metrics-server/pull/182).

Comment (deniseschannon) [reply recipes/metrics-server/metrics-server-deployment.yaml]: can you switch this to `rancher/metrics-server:v0.3.1` as we package it as multi-arch and we can use that until upstream adds in multi-arch.

Comment (meldafrawi) [reply recipes/metrics-server/metrics-server-deployment.yaml]: @deniseschannon 

I tried to test rancher metrics-server multi-arch image you mentioned on the same raspberry pi, but i got an error stating that:
`no matching manifest for unknown in the manifest list entries`

The raspberry pi architecture is **armv7l**

Comment (erikwilson) [reply recipes/metrics-server/metrics-server-deployment.yaml]: Thanks for trying it out @meldafrawi! Is it possible for us to add armv7 support to rancher/metrics-server @deniseschannon?

Comment (meldafrawi) [reply recipes/metrics-server/metrics-server-deployment.yaml]: @erikwilson, You're welcome.

Comment (deniseschannon) [reply recipes/metrics-server/metrics-server-deployment.yaml]: @erikwilson we'd have to sync up internally that created this image and create PRs to be able to do that. can you reach out to see who created it?

Comment (erikwilson) [reply recipes/metrics-server/metrics-server-deployment.yaml]: /ping @loganhz

Comment (loganhz) [reply recipes/metrics-server/metrics-server-deployment.yaml]: We don’t have a arm7 drone box.
So our CI only has amd64 build and arm64 build.

We need to talk to Darren and Bill about it. @erikwilson

Comment (ibuildthecloud) [reply recipes/metrics-server/metrics-server-deployment.yaml]: @loganhz we do have arm7.  k3s CI uses it.  You just need to put `arm` as the architecture in .drone.yaml.  It's technically a ARM64 VM but running a armv7 version of docker which pulls armv7 architecture images.

Comment (loganhz) [reply recipes/metrics-server/metrics-server-deployment.yaml]: Sorry. Just recalled. @cjellick helped to pushed the metrics server images manually to rancher dockerhub for the ARM64 Support. 
https://github.com/jianghang8421/publish-manifest/blob/master/publish-manifest-images.sh#L102-L112

So we don't have a CI build for rancher/metrics-server images currently. (Good to know we have arm7 build!)

We need someone to pull `gcr.io/google_containers/metrics-server-arm:v0.3.1`, retag and update the `rancher/metrics-server` manifest images manually.

Comment (ibuildthecloud) [reply recipes/metrics-server/metrics-server-deployment.yaml]: I'll look into this, @loganhz thanks

Comment (rcarmo) [reply recipes/metrics-server/metrics-server-deployment.yaml]: Guys, I do Travis builds for multiple ARM architectures using QEMU,. Check out https://github.com/insightfulsystems/alpine-python/blob/master/.travis.yml

**Comment by staranto** [comment]
Any update @erikwilson @ibuildthecloud ?  

I've got a multi-arch k3s v0.8.0 cluster.  There are five arm64 nodes and one amd64 node.  On a **arm64** node ... 
```
> crictl pull rancher/metrics-server:v0.3.1-arm64
    Image is up to date for sha256:3f2f430f6b7b2e0575db9b896a060c6238230bdb2f8c6c55d3ca21adc93add8c 
> crictl inspecti rancher/metrics-server:v0.3.1-arm64 | jq ".info.imageSpec.architecture"
"amd64"
```
Pulling the "official" image (gcr.io/google_containers/metrics-server-arm64:v0.3.1) exhibits the same problem as it is identical to Rancher's.

In the end, I'm stuck...  I can't seem to find a way to deploy metrics-server to an arm64 cluster.  I could live with not deploying it to the single amd64 node.

Any ideas?  I feel like I'm missing something obvious.  

Thanks!!

**Comment by erikwilson** [comment]
Waiting for upstream kubernetes-incubator/metrics-server/pull/313 to fix kubernetes-incubator/metrics-server/issues/181

**Comment by erikwilson** [comment]
Fixed with #1015

**Pull request closed without merging**